README.md

```diff
@@ -1,39 +1,597 @@
 
-Add this line to your application's Gemfile:
+# Table of contents
+
+- [Installation](#installation)
+- *[Importers](#importers): [ActiveRecord](#activerecord-importer), [Avro](#avro-importer), [CSV](#csv-importer), [Excel](#excel-importer), [Excelx](#excelx-importer), [HTML](#html-importer), [JSON](#json-importer), [Mongo](#mongo-importer), [Plaintext](#plaintext-importer), [RData](#rdata-importer), [RDS](#rds-importer), [Redis](#redis-importer), [SQL](#sql-importer)*
+- *[Exporters](#exporters): [Avro](#avro-exporter), [CSV](#csv-exporter), [Excel](#excel-exporter), [JSON](#json-exporter), [RData](#rdata-exporter), [RDS](#rds-exporter), [SQL](#sql-exporter)*
+- [Tweaking](#tweaking-io-modules)
+- [Contributing](#contributing)
+- [License](#license)
+
+# Installation
+
+[(Go to Table of Contents)](#table-of-contents)
+
+- If you're working with a Gemfile,
+
+    - Add this line to your application's Gemfile:
+
+        ```ruby
+        gem 'daru-io'
+        ```
+
+    - And then execute on your terminal:
+
+        ```sh
+        bundle
+        ```
+
+- If you're NOT working with a Gemfile, simply install it yourself by executing on your terminal:
+
+    ```sh
+    gem install daru-io
+    ```
+
+- Require `daru-io` gem in your application:
+
+    ```ruby
+    require 'daru/io' #! Requires all Importers & Exporters
+    require 'daru/io/importers' #! Requires all Importers and no Exporters
+    require 'daru/io/importers/json' #! Requires only JSON Importer
+    ```
+
+**Note: Each IO module has it's own set of dependencies. Have a look at the [Importers](#importers) and [Exporters](#exporters) section for dependency-specific information.**
+
+# Importers
+
+The **Daru::IO** Importers are intended to return a **Daru::DataFrame** from the given arguments. Generally,
+all Importers can be called in two ways - from **Daru::IO** or **Daru::DataFrame**.
 
 ```ruby
-gem 'daru-io'
+#! Partially requires Format Importer
+require 'daru/io/importers/format'
+
+#! Usage from Daru::IO
+Daru::IO::Importers::Format.new(args).call
+
+#! Usage from Daru::DataFrame
+Daru::DataFrame.from_format(args)
 ```
 
-And then execute:
+**Note: Please have a look at the respective Importer Doc links below, for having a look at arguments and examples.**
+
+### ActiveRecord Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+Imports a **Daru::DataFrame** from an **ActiveRecord** connection.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/ActiveRecord)
+- **Gem Dependencies**: `activerecord` gem
+- **Other Dependencies**: None
+- **Usage**:
+    ```ruby
+    #! Partially require just ActiveRecord Importer
+    require 'daru/io/importers/active_record'
+
+    #! Usage from Daru::IO
+    Daru::IO::Importers::ActiveRecord.new(args).call
+
+    #! Usage from Daru::DataFrame
+    Daru::DataFrame.from_activerecord(args)
```

Comment:
Let's use more realistic examples, with meaningful arg.names. The idea that for simplest usage it should be possible to just copy-paste the example from README and go with it.